### PR TITLE
Copy query settings correctly

### PR DESCRIPTION
**Pull request by KanishManuja-MS**
### Issues

Setting DefaultQuerySettings using the extensions on the route builder does nothing. 

### Description

Now, we will respect the default query settings. 

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.AspNet.OData.Shared/Query/DefaultQuerySettings.cs [hunk 1]
```diff
@@ -64,5 +64,16 @@ public class DefaultQuerySettings
         /// Gets or sets a value indicating whether the service will use skiptoken or not.
         /// </summary>
         public bool EnableSkipToken { get; set; }
+
+        internal void CopySettings(DefaultQuerySettings settings)
```
Comment (xuzhg) [inline src/Microsoft.AspNet.OData.Shared/Query/DefaultQuerySettings.cs:68]: Clone?

File: src/Microsoft.AspNetCore.OData/Extensions/ODataServiceProviderConfigExtenions.cs [hunk 2]
```diff
@@ -34,7 +34,7 @@ public static void SetDefaultQuerySettings(this IServiceProvider serviceProvider
                 throw Error.InvalidOperation(SRResources.MissingODataServices, nameof(DefaultQuerySettings));
             }
 
-            querySettings = defaultQuerySettings;
+            querySettings.CopySettings(defaultQuerySettings);
```
Comment (mikepizzo) [inline src/Microsoft.AspNetCore.OData/Extensions/ODataServiceProviderConfigExtenions.cs:37]: So we are overwriting whatever settings were obtained from Dependency Injection with the settings passed in to SetDefaultQuerySettings. Is this the expected behavior?

Note that we are copying MaxTop from the default settings returned from DI *after* we have already applied semantics based on the value passed in (see line 26).  Should we move the logic on line 26-29 to *after* we do copysettings?

File: src/Microsoft.AspNet.OData.Shared/Query/DefaultQuerySettings.cs [hunk 3]
```diff
@@ -64,5 +64,16 @@ public class DefaultQuerySettings
...
+        {
+            this.EnableCount = settings.EnableCount;
+            this.EnableExpand = settings.EnableExpand;
+            this.EnableFilter = settings.EnableFilter;
+            this.EnableOrderBy = settings.EnableOrderBy;
+            this.EnableSelect = settings.EnableSelect;
+            this.EnableSkipToken = settings.EnableSkipToken;
+            this.MaxTop = settings.MaxTop;
```
Comment (mikepizzo) [inline src/Microsoft.AspNet.OData.Shared/Query/DefaultQuerySettings.cs:76]: Should we only copy this over if this.MaxTop.HasValue == true, or would we need to copy it over in order to allow clearing a MaxTop through DI?

File: src/Microsoft.AspNetCore.OData/Extensions/ODataServiceProviderConfigExtenions.cs (hunk 2 above)
Comment (KanishManuja-MS) [reply src/Microsoft.AspNetCore.OData/Extensions/ODataServiceProviderConfigExtenions.cs:32]: Yes. Looking at the name of the extension method it should reset the default query settings with the current object. 

Moved the three lines after copying.

File: src/Microsoft.AspNet.OData.Shared/Query/DefaultQuerySettings.cs (hunk 3 above)
Comment (KanishManuja-MS) [reply src/Microsoft.AspNet.OData.Shared/Query/DefaultQuerySettings.cs:76]: I think we need to copy to reset the MaxTop to even null.

**Comment by xuzhg** [comment]
Merged at: https://github.com/OData/WebApi/commit/4d8c2baeaf5bdbe650d5b7d80b0753f7efba6229

**Pull request closed without merging**